Refactor/cleanup

fayer3 opened the pull request:
brings back old variable names and formatting
adds parchment mappings (closes #189)
adds mixin extras (not yet done completely)
ads javadoc to most methods

some of the other changes/fixes that happened:
fix seated crawl head floating
remove tpose from mcvr, use the null provider instead for that
removes MCVR.getHardwareType (since it is unused)
renames `MCVR.hasThirdController` to `hasCameraTracker`, to reflect what it does
fix key simulation with modifiers
remove vsync change buffer reinit
fix menu tooltips being stuck with keyboard use when the button disapears

move all shader rendering to `org/vivecraft/client_vr/render/helpers/ShaderHelper`, includes FSAA, mixed reality and VR post/fov reduction
removed getFrameTiming/getCurrentTimeSecs from VRRenderer (unnecessary, replaced the one call that used it with Util.getMillis)
removed useless overrides from OpenVRStereoRenderer
move inMenuroom methods out of gamerenderer to MethodHolder

change logger in MixinConfig to a separate one

moved lwjgl library check to MCOpenVR
removed MCVR.postInit, since there is no point in having it not in the regular init anymore
move touchhotbar after pose update, to not be a frame behind

fix mirror aspect for third and center view
fix black in block overlay, not blocking with wide aspects
fix interactive wrist hotbar with changed main hand, and offset with reversed hands

remove options reload from mcopenvr init, and move the external camera config loading to the vr settings init

update editor config to allow one line ifs and specify how if wrapping should be like 

fix auto open keyboard setting
fix radial/keaboard guiscale issue with max hud scale
add option to toggle headshot hitboxes

fixes #277 
fixes #270
fixes #267
fixes #264
fixes #250

fayer3 committed:
add parchment mappings
add mixin extras
add back old names and comments
add some javadoc

some fixes/remove of unused stuff:
fix seated crawl head floating
remove tpose from mcvr, use the nullprovider for that
renames `MCVR.hasThirdController` to `hasCameraTracker`, to reflect what it does
fix key simulation with modifiers
remove vsync change buffer reinit
fix menu tooltips being stuck with keyboard use when the button disappears

fayer3 committed:
refactor part 2:
move shaders into a shader helper
includes FSAA, mixed reality and VR post/fov reduction
move inMenuroom methods out of gamerenderer
make ActionParams a record
bunch of javadoc
re sort the methods in MCVR
fix thirdperson aspect being stuck at 16x9

fayer3 committed:
refactor/cleanup part 3:
more javadoc and variable renames
moved lwjgl library check to MCOpenVR
removed MCVR.postInit, since there is no point in having it not in the regular init anymore
move touch hotbar after pose update, to not be a frame behind
remove options reload from MCOpenVR init, and move the external camera config loading to the vr settings init
change ActionParams back to regular class for backwards compatibility

ferriarnus commented on common/src/main/java/org/vivecraft/client_vr/gameplay/VRPlayer.java:
```diff
@@ -249,208 +338,235 @@ public void setRoomOrigin(double x, double y, double z, boolean reset) {
 
-            this.doPlayerMoveInRoom(player);
+        if (!this.initDone) {
+            System.out.println("<Debug info start>");
```
this should also use the logger no?

fayer3 replied:
oops missed that, but this is still a work in prograss, ther are still many system outs I think

ferriarnus commented on common/src/main/java/org/vivecraft/client_vr/gameplay/VRPlayer.java:
```diff
@@ -476,92 +598,116 @@ public boolean getFreeMove() {
-                if (!(itemstack.getItem() instanceof CrossbowItem crossbowitem)) {
-                    break label54;
-                }
+        if (itemStack.getItem() instanceof SnowballItem ||
```
seems we forgot our tag here?

fayer3 replied:
not sure if the crossbow item tag would work there, since it actually needs to be a crossbow item for the charge check I think, unless you mean separate from the charge check

ferriarnus replied:
I mean `itemstack.is(ItemTags.VIVECRAFT_THROW_ITEMS)`

fayer3 replied:
ah

fayer3 committed:
vr palyer logger and THROW tag check

fayer3 committed:
removes legacy profiles from the settings
moves the settings file to the modloaders config folder

fayer3 committed:
check for settings file existence before reading
some logging fixes

fayer3 committed:
fix common resources in dev missing
remove legacy resource extraction
some logging fixes

fayer3 committed:
remove ScreenUtils, use vanilla implementation instead

fayer3 committed:
don't import any full packages by default

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.20.4' into refactor-cleanup

fayer3 committed:
more javadoc ans variable names

fayer3 committed:
change RenderConfigException and the ErrorScreen to use a Component as title

fayer3 committed:
move vrInitialized back before vrrenderer setup, to fix an iris issue

fayer3 committed:
limit the dh shader reload fix to when DH is loaded

fayer3 committed:
add option to use mipmaps on gui, and only generate them when that is used
fix Keyboard/Radial buffer not resizing with the 1440p gui option

fayer3 committed:
fix potential memory leak, when toggling vr with a post effect active
fix missing vanilla post effect, when disabling vr

fayer3 committed:
more javadoc and variable renames

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.20.4' into refactor-cleanup

fayer3 committed:
more javadoc and variable names
change climbey blocklist command to check for valid blocks
move the method to send stuff to all seen players into its own method, to not have duplicated code there

fayer3 committed:
change climbeyBlocklist to allow invalid blocks in the config, to not reset everything
warns in the log if something is invalid, and only sends valid blocks to clients

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.20.4' into refactor-cleanup

fayer3 committed:
remove player argument from ServerVivePlayer methods

fayer3 committed:
javadoc/formatting

fayer3 committed:
javadoc for Xplat/Xevents
remove Xplat tick start/end events, since they are not used/needed

ferriarnus commented on common/src/main/java/org/vivecraft/client/gui/framework/VROptionLayout.java:
```diff
@@ -28,94 +28,63 @@ public class VROptionLayout {
         this._ordinal = ordinal;
+    }
+
+    public VROptionLayout(Position pos, float row, boolean enabled, String title) {
```
Maybe at nullable to the title, or simply pass "" instead of null.

ferriarnus commented on common/src/main/java/org/vivecraft/client/network/ClientNetworking.java:
```diff
@@ -221,7 +224,7 @@ public static void handlePacket(CommonNetworkHelper.PacketDiscriminators packetI
         Minecraft mc = Minecraft.getInstance();
         switch (packetID) {
             case VERSION -> {
-                String s11 = buffer.readUtf(1024);
```
this `readUtf(...)` should be possible to not have hardcoded like this? Maybe a future rework.

fayer3 replied:
so remove the length limit? can do that

ferriarnus replied:
Looking at it again now, this may not matter. The default in vanilla is much higher (32767). As such it's probably fine to keep? My original comments was me wondering why we need such a big buffer, but it seems vanilla uses an even bigger one. If we know the size of the string (or know it's smaller then 1024), keeping this is the better option it seems. I am still wondering where these "magic" numbers come from.

ferriarnus commented on common/src/main/java/org/vivecraft/client/utils/Utils.java:
```diff
@@ -334,219 +340,158 @@ private static void handleAssetException(Throwable e, String name, boolean requi
         }
     }
 
     public static void unpackNatives(String directory) {
```
how many of these aren't needed anymore with how we do natives now?

fayer3 replied:
not sure, threadmill natives still need to be extracted I think

fayer3 replied:
cleaned up that class, removed some unused stuff, and split it into more specific sub classes

fayer3 committed:
more javadoc

fayer3 committed:
mixin extensions javadoc,
rename PlayerExtension.setItemInUseCountClient to setItemInUseRemainingClient, to correspond to the variable name
rename some partialTicks to partialTick
remove FrustumExtension.vivecraft$setCameraPosition(), use Frustum.prepare() instead, that does the same

fayer3 committed:
fix NullVR not counting the headset as tracked

fayer3 committed:
parameter rename that was forgotten in the previous commit

fayer3 committed:
move the movementTeleportTimer to the teleportTracker

fayer3 committed:
move blit shader call out of rendertarget mixin
add custom blit shader
add _vr suffix to all shaders
rename fovreduction shader to postprocessing

fayer3 committed:
delete old shader files

fayer3 committed:
use a regular ModifyArg to change filtering mode

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.20.4' into refactor-cleanup

fayer3 committed:
fix auto open keyboard setting, also add a chat option, to have the previous behavior
re add the "Click to open keyboard" hint to text fields

fayer3 committed:
change option parser to system properties, to be able to set them with the jvm args in most launchers

fayer3 committed:
some mixines cleanup
make blending overrides use ModifyArg

fayer3 committed:
window mixin cleanup

fayer3 committed:
fix radial/keyboard max gui scale issue, fixes #264

fayer3 committed:
fix RenderSystem mixin injection failing

fayer3 committed:
update arch loom to 1.6, and gradle to 8.6

fayer3 committed:
move isClimbeyClimbEquipped to the ClimbTracker
make Trackers item checks static methods
renamed `isClimbeyXEquipped` to `hasClimbeyXEquipped` since it checks for the provided Player now

fayer3 committed:
use ModifyExpressionValue for liquid block renderer wrapping override

fayer3 committed:
cleanup more mixins
add a setting to not render head hitboxes always
remove unneeded unused code for vivecraft item colors

fayer3 committed:
make RenderLayerTypes directly the enum

fayer3 committed:
make already gone through mixins all private if they don't need to be public

fayer3 committed:
fix wrong function call breaking fabulous menus

ferriarnus reviewed (commented):
just some questions

fayer3 committed:
fix crumbling effect rendering in fabulous with iris loaded

fayer3 committed:
fix GuiGraphics blending mixin

fayer3 committed:
change GuiGraphics mixin to actual class name

fayer3 committed:
screen mixins cleanup

fayer3 committed:
cleanup pause menu mixins,
use mixin extras instead of redirect
remove unnecessary/broken button disable for the report bugs button
add setting to disable pause menu modifications

fayer3 committed:
fix method visibility of the previous commits

fayer3 committed:
refactor gui mixin,
move crosshair rendering to RenderHelper
use WrapOperation to change the offhand slot color

fayer3 committed:
fix pressing shift in gui blocking stack splitting in nonvr

fayer3 committed:
cleanup Tutorial toast mixin, and use a Shared variable instead of the unique variable

fayer3 committed:
add methods to get yaw/pitch of VRData directly in radians
change RunTracker yaw to radians
use WrapOperation instead of redirects in ItemPickupParticleVRMixin
more cleanups

fayer3 committed:
merge EntityRenderDispatchers getCameraOrientationOffset and cameraOrientation into one getVRCameraOrientation with scale and offset parameter
make cameraOrientation mixin not as invasive, and just don't do anything when not in a vr pass
same for EntityRendererVRMixin.cameraOffset

fayer3 committed:
fix VRPlayerRenderer heightscale weirdness

fayer3 committed:
fix fishing line positioned wrong for other vr players

fayer3 committed:
cleanup and switch to wrap operation

fayer3 committed:
remove invalid maven repo

fayer3 committed:
add mixin extras to forge

fayer3 committed:
cleanup GameRendererVRMixin
- fix projection matrix not updating on render distance change
- remove unneeded override of the far clipping plane
- remove unnecessary resets at beginning of rendering
- fix possibility of hitting your own vehicle

fayer3 committed:
add some support for mods that alter gui scale, fixes #267

fayer3 committed:
ItemInHandRenderer cleanup

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.20.4' into refactor-cleanup

fayer3 committed:
cleanup from last merge

fayer3 committed:
only set custom stencil when the modloader doesn't support it

fayer3 committed:
make nausea mixin more change proof

fayer3 committed:
fix missed thing from last merge

fayer3 committed:
fix cropped mirror not accounting for the source aspect ratio

fayer3 committed:
more cleanup

fayer3 committed:
update arch loom and deps and move lwjgl version to gradle.properties

fayer3 committed:
cleanup doors and boat
limit roomscale boat speed to vanilla value, and add gradual turning when pointing

fayer3 committed:
make boat steer deadzone bigger again, and respect the hmd freemove setting

fayer3 committed:
remove quiltflower since vineflower is now included with loom

fayer3 committed:
fix gui cursor on chat screen with journeymap

fayer3 committed:
handle wonder food with an alteration of can eat, instead of a separate inject, and some cleanup

fayer3 committed:
cleanup key handling
remove broken cancellation of key handling when right control is held down, don't think that is needed
add method to MCVR to handle custom key inputs, and use it for the NullVR hotswitch
use a wrapOperation for the screenshot trigger
fix broken keyboard closing with autoCloseKeyboard disabled

fayer3 committed:
run import optimizer

fayer3 committed:
don't completely cancel the keyboard input tick

fayer3 committed:
Options and MouseHandler

fayer3 committed:
cleanup MinecraftVRMixin
move current arm swing type to ClientDataHolderVR
fix attack to the air not updating the arm swing type
move mirror notification to its own class
handle mixed reality keys with the other VRHotkeys
only tick menuWorldRenderer when in the menu

fayer3 committed:
fix ItemPickupParticle offset

fayer3 committed:
cleanup server mixins

fayer3 committed:
fix eating particles for vr players
fix sweeping particles for vr players

fayer3 committed:
fix ServerVivePlayers ServerPlayer reference getting out of syn on respawn

fayer3 committed:
redo enderman stuff to not cancel, and modify headpos for any `hasLineOfSight` call, not just from enderman
use the bigger visibility cone from the spigot extension
don't cancel creeper swell goal, just change the distance check

fayer3 committed:
rename callbackinfos missed previously

fayer3 committed:
cleanup projectiles
fix arrow speed ignoring roomscale bow draw

fayer3 committed:
cleanup crossbow

fayer3 committed:
cleanup mod compat mixins
move mod loaded checks to their Helper classes
fix dynamicFPS mod getting quieter when window is unfocused in vr
simplify NonNull redirects for the iris compat
fix iris/oculus sometimes disabling AO in the menuworld
remove player repositioning in optifine mixin. that is already done in levelrenderer
fix resolution control 3 compat
remove sodium options mixin for the graphic mode change, it is already checked every frame in the VRRenderer
make all helper reflection variables match, with Class_Method/Field naming

fayer3 committed:
use annotation to determine needed classes for a mixin

fayer3 committed:
remove legacy telescope models

fayer3 committed:
only process vr hotkeys when vr is initialized, fixes #299
fix debug winscreen not close able

fayer3 committed:
fix not being able to click on the left parts of the text fields on SettingsLists

fayer3 committed:
java doc for WidgetBuilder

fayer3 committed:
reformat long ifs

fayer3 committed:
add Vivecraft to all log messages

fayer3 committed:
add this keyword to all class variables

fayer3 committed:
fix MixinConfig

fayer3 committed:
fix haptics on death not triggering

fayer3 committed:
fix last few mixins

fayer3 committed:
forgot those

fayer3 committed:
network redo and add enum server config value

fayer3 committed:
use mixins instead of overrides in the LocalPlayer

fayer3 committed:
fix some mixins, move mixin extras version to gradle.properties

fayer3 committed:
add nullabel to VROptionLayout title, and consistently use the javax nullable

fayer3 committed:
remove some unneeded file utils and use Commons IO IOUtils where applicable

fayer3 committed:
split Utils class into general, text, math and file Utils

fayer3 committed:
fix mistake in javadoc

fayer3 committed:
fix clicking reset origin with vr off crashing the game
fix pressing escape on the seated confirmation closing the screen, instead of canceling the confirmation

fayer3 committed:
fix vr active packet having wrong packet identifier

fayer3 committed:
fix texture animation in the main menu, vanilla always ticks the texture manager since 1.20.2

fayer3 committed:
fix play area size detection

fayer3 committed:
fix camera config loading crashing on startup

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.20.4' into refactor-cleanup

fayer3 committed:
cleanup of merge

fayer3 committed:
make all post effects toggle able, fix portal effect

fayer3 committed:
fix crash with non existing server config

fayer3 committed:
fix analogue movement

fayer3 committed:
fix typo breaking optifine

fayer3 committed:
move pehkui scaling into a general ScaleHelper for future mc compat

fayer3 committed:
move mirror copy to ShaderHelper

ferriarnus commented on common/src/main/java/org/vivecraft/client_vr/Vec3History.java:
```diff
@@ -7,126 +7,157 @@
 public class Vec3History {
-    private final int _capacity = 450;
-    private final LinkedList<entry> _data = new LinkedList<>();
+    private static final int _capacity = 450;
```
Hmm, lets maybe not use the _ convention? Also a "this" is removed in this file, which feels odd

fayer3 replied:
if you mean the this for _capacity, that is because I made that field static

ferriarnus replied:
Don't we use CAPS for static?

fayer3 replied:
not sure, I can change all static fieds to caps though
all static, or only static final?

ferriarnus commented on common/src/main/java/org/vivecraft/client_vr/Vec3History.java:
```diff
@@ -7,126 +7,157 @@
+    /**
+     * Entry holding a position and timestamp
+     */
+    private static class Entry {
```
For newer java versions this *could* be a record instead

fayer3 replied:
we could yeah, would just mean that if we ever backport to pre 1.17 there will be differences, if that doesn't matter, I can change some stuff to recrods

ferriarnus replied:
It doesn't really matter, but then again pre 1.17 seems unlikely. Just a note that we can use records for classes like this. I'm fine with keeping it.

ferriarnus commented on common/src/main/java/org/vivecraft/client_vr/Vec3History.java:
```diff
@@ -7,126 +7,157 @@
+    /**
+     * Entry holding a position and timestamp
+     */
+    private static class Entry {
```
For newer java versions this *could* be a record instead

ferriarnus commented on common/src/main/java/org/vivecraft/client_vr/provider/openvr_lwjgl/MCOpenVR.java:
```diff
@@ -584,286 +674,296 @@ private void generateActionManifest() {
+            this.controllerComponentTransforms.containsKey(componentName) &&
+            this.controllerComponentTransforms.get(componentName)[controllerIndex] != null ?
+            this.controllerComponentTransforms.get(componentName)[controllerIndex] :
+            new Matrix4f();
```
setIdentity not needed?

fayer3 replied:
a new Matrix4f object is already set to identity in the constructor, not sure why there was a second identity

ferriarnus commented on common/src/main/java/org/vivecraft/client_vr/render/VRShaders.java:
```diff
@@ -5,27 +5,29 @@
+
+    // mixed reality shader and its uniforms
+    public static ShaderInstance mixedRealityShader;
+    public static AbstractUniform _MixedReality_hmdViewPosition;
```
Here as well, the  _ convention isn't something we use, and also seems to be missing some "this" before the variables

ferriarnus reviewed (commented):
It looks good, but it's a bit much for my eyes 😅

fayer3 committed:
delete unused classes

fayer3 committed:
use CAPS for static fields
use record classes for data holders

fayer3 committed:
only send vivecraft packets, if the server sent a VERSION packet
remove client specific case for climbing tracker

fayer3 committed:
fix trapdoors with ladders not climbable

fayer3 committed:
forgot some CAPS

fayer3 committed:
missing this
